### ISSUE TITLE
Add warm up and cool dowm time also on setting up a teleporting location but not only teleporting to it.

### ISSUE THREAD

**Issue by Ch3225** [WiIIiam278/HuskHomes2#290]
When a player died, I don't want them to use /back command to teleport to their death location. I just turn off the /back command. 
But the problem comes. Actually players can use the command like /sethome to set a teleport location when they are about to die(for example, when a player is drowning, being surrounded by monsters, etc.) They have no method to escape at that time, but they can set teleport location there, so even I have banned /back command they can still use this trick to directly teleport to their death location). 
Also, players may use sethome too often when they are searching/mining to get back to their locations. If they use that too often... It's not so interesting yeah?
I wanna prevent this from happening, so I come up with the solution of also adding warm up and cool down time on setting teleport locations but not only when players are about to teleport.
Actually there seems to be an alternative solution, which is adding a cost when they are setting teleport .But you know, in many games, like MOBA, skills with a cool down time and something that "could be able to buy and consume" are two sort of things.

**Comment by WiIIiam278** [comment]
Well, I see what you're saying--certainly a good point and this is the sort of "cheese" that's existed for awhile in similar systems upon which HuskHomes is based.

I think your proposal may get annoying for users in certain circumstances tbh at the expense of added code complexity where I'd rather avoid this.

I suggest looking into catch-all command cool-down plugins :/)

**Comment by Ch3225** [comment]
Thank you for your advice.
You are right. For the annoying on users.... you give me a good warning. It might annoying players some way. My server is not a pvp server, and I used a plugin called Angelchest. I also use EliteMobs. Many Minecraft players are not newbies so I wanna give them some challanges. When player died, they will be sent a coordinate of their death allowing them getting their stuffs again, from somewhere like their camp. But you are right...I think again, we should not overdo. Maybe someone would died over and over again and lost all the stuff... Then he will feel bad and leave the server.
But for the expense of added code complexity... It's been a long time that I haven't use Java but when I learned Java, I learned a thing called AOP....I think it could help....But anyway, that feature is not that worth to develop.
Maybe I should start to write some Java again.
Anyway, thank you for advice and your plugin. I've been using your plugin on your server for a while. It is really a cool plugin.

**Comment by Ch3225** [comment]
Thank you for your advice.
You are right. For the annoying on users.... you give me a good warning. It might annoying players some way. My server is not a pvp server, and I used a plugin called Angelchest. I also use EliteMobs. Many Minecraft players are not newbies so I wanna give them some challanges. When player died, they will be sent a coordinate of their death allowing them getting their stuffs again, they have to move up from somewhere like their camps. But you are right...I think again, we should not overdo. Maybe someone would died over and over again and lost all the stuff... Then he will feel bad and leave the server.
But for the expense of added code complexity... It's been a long time that I haven't use Java but when I learned Java, I learned a thing called AOP....I think it could help....But anyway, that feature is not that worth to develop.
Maybe I should start to write some Java again.
Anyway, thank you for advice and your plugin. I've been using your plugin on your server for a while. It is really a cool plugin.
